File: Java_Selenium_Automation/src/test/java/com/aiu/pages/LoginPage.java
package com.aiu.pages;

import com.aiu.utils.ConfigReader;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

/**
 * LoginPage - Page Object for Login page
 */
public class LoginPage extends BasePage {

    // Locators
    @FindBy(id = "email")
    private WebElement emailInput;

    @FindBy(id = "password")
    private WebElement passwordInput;

    @FindBy(xpath = "//button[@type='submit']")
    private WebElement loginButton;

    @FindBy(xpath = "//div[contains(@class, 'error') or contains(@class, 'alert')]")
    private WebElement errorMessage;

    @FindBy(linkText = "Register")
    private WebElement registerLink;

    @FindBy(linkText = "Forgot Password?")
    private WebElement forgotPasswordLink;

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    /**
     * Navigate to login page
     */
    public void open() {
        navigateTo(ConfigReader.getBaseUrl() + "/login");
        logger.info("Opened Login page");
    }

    /**
     * Enter email
     */
    public void enterEmail(String email) {
        type(emailInput, email);
        logger.info("Entered email: " + email);
    }

    /**
     * Enter password
     */
    public void enterPassword(String password) {
        type(passwordInput, password);
        logger.info("Entered password");
    }

    /**
     * Click login button
     */
    public void clickLoginButton() {
        click(loginButton);
        logger.info("Clicked login button");
    }

    /**
     * Perform complete login
     */
    public void login(String email, String password) {
        enterEmail(email);
        enterPassword(password);
        clickLoginButton();
        waitForPageLoad();
        logger.info("Performed login with email: " + email);
    }

    /**
     * Login with admin credentials
     */
    public void loginAsAdmin() {
        login(ConfigReader.getAdminEmail(), ConfigReader.getAdminPassword());
    }

    /**
     * Login with student credentials
     */
    public void loginAsStudent() {
        login(ConfigReader.getStudentEmail(), ConfigReader.getStudentPassword());
    }

    /**
     * Login with organizer credentials
     */
    public void loginAsOrganizer() {
        login(ConfigReader.getOrganizerEmail(), ConfigReader.getOrganizerPassword());
    }

    /**
     * Check if error message is displayed
     */
    public boolean isErrorDisplayed() {
        return isDisplayed(errorMessage);
    }

    /**
     * Get error message text
     */
    public String getErrorMessage() {
        return getText(errorMessage);
    }

    /**
     * Click register link
     */
    public void clickRegisterLink() {
        click(registerLink);
    }

    /**
     * Check if on login page
     */
    public boolean isOnLoginPage() {
        return getCurrentUrl().contains("/login");
    }
}

File: Selenium_Automation_Tests/pages/login_page.py
"""
Login Page Object
"""

from selenium.webdriver.common.by import By
from pages.base_page import BasePage
from utils.config import Config


class LoginPage(BasePage):
    """Login page object model"""
    
    # Locators
    EMAIL_INPUT = (By.ID, "email")
    PASSWORD_INPUT = (By.ID, "password")
    LOGIN_BUTTON = (By.XPATH, "//button[contains(text(), 'Login') or contains(text(), 'Sign In')]")
    ERROR_MESSAGE = (By.CSS_SELECTOR, ".error-message, .alert-error, [role='alert']")
    FORGOT_PASSWORD_LINK = (By.LINK_TEXT, "Forgot Password")
    REGISTER_LINK = (By.LINK_TEXT, "Register")
    
    def __init__(self, driver):
        super().__init__(driver)
        self.url = Config.get_url('/login')
    
    def open(self):
        """Navigate to login page"""
        super().open(self.url)
    
    def login(self, email, password):
        """
        Perform login action
        
        Args:
            email (str): User email
            password (str): User password
        """
        self.enter_email(email)
        self.enter_password(password)
        self.click_login_button()
    
    def enter_email(self, email):
        """Enter email"""
        self.enter_text(self.EMAIL_INPUT, email)
    
    def enter_password(self, password):
        """Enter password"""
        self.enter_text(self.PASSWORD_INPUT, password)
    
    def click_login_button(self):
        """Click login button"""
        self.click(self.LOGIN_BUTTON)
    
    def get_error_message(self):
        """Get error message text"""
        return self.get_text(self.ERROR_MESSAGE)
    
    def is_error_displayed(self):
        """Check if error message is displayed"""
        return self.is_element_visible(self.ERROR_MESSAGE, timeout=3)
    
    def click_forgot_password(self):
        """Click forgot password link"""
        self.click(self.FORGOT_PASSWORD_LINK)
    
    def click_register(self):
        """Click register link"""
        self.click(self.REGISTER_LINK)
